ui/app/pages/permissions-connect/permissions-connect.container.js

```diff
@@ -42,6 +42,15 @@ const mapStateToProps = (state, ownProps) => {
   const { origin } = metadata
   const nativeCurrency = getNativeCurrency(state)
 
+  const domainMetadata = getDomainMetadata(state)
+  const targetDomainMetadata = origin
```

Comment:
Nit: Rather than this being re-calculated each render but only used once, you could move this to `permissions-connect.component` to the `state` initialization so that it's only calculated in the one place where it's used.

Reply:
Oh wait - it _is_ used, to detect changes in the name. Nevermind then!